feat!: v1alpha2 release

devotional-phoenix-97 opened the pull request:
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Other (please describe):

## What is the current behavior?

- We don't know how to merge v1alpha2 to main

## What is the new behavior?

<!-- Please describe the behavior or changes that are being added by this PR. -->

- We periodicallymerge main and v1alpha2
- We know how merge into main

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

AxiomaticFixedChimpanzee committed:
feat: scaffold v1alpha2 of Paas CRD (#508)

* feat(api): scaffold v1alpha2 of Paas

* feat(api): set storage version for Paas

* feat(api): define fields of v1alpha2 Paas version

* feat(api): scaffold Paas version conversion functions

* chore: fix linting errors

* feat(v1alpha2): add fields to namespace map

* test(v1alpha2): scaffold conversion test

feat(v1alpha2): implement conversion to and from v1alpha1 (#509)

* feat: scaffold v1alpha2 of Paas CRD (#508)

* feat(api): scaffold v1alpha2 of Paas

* feat(api): set storage version for Paas

* feat(api): define fields of v1alpha2 Paas version

* feat(api): scaffold Paas version conversion functions

* chore: fix linting errors

* feat(v1alpha2): add fields to namespace map

* test(v1alpha2): scaffold conversion test

* feat(v1alpha2): implement conversion to v1alpha1

* chore: remove namespaced from Paas kubebuilder config

* feat(v1alpha2): implement conversion from v1alpha1

* feat: implement conversion webhook for Paas v1alpha1 to v1alpha2

A Paas can now be posted in either version, and the conversion from/to
the storage version (v1alpha1) will happen automatically. Similarly,
conversion will occur transparently when requesting v1alpha2.

fix: git custom field keys for Paas

test(e2e): conversion between Paas versions

devotional-phoenix-97 committed:
Merge branch 'main' into v1alpha2

devotional-phoenix-97 committed:
Merge branch 'main' into v1alpha2

devotional-phoenix-97 committed:
chore: Fixing revive/exported issues for v1alphav2 (#530)

* Fixing revive/exported issues for v1alphav2

* Updating manifests

devotional-phoenix-97 committed:
adding v1alpha2 ginkgo (#531)

devotional-phoenix-97 committed:
adding v1alpha2.paasns (#536)

devotional-phoenix-97 committed:
Merge branch 'main' into v1alpha2

AxiomaticFixedChimpanzee committed:
Merge branch 'main' into v1alpha2

AxiomaticFixedChimpanzee committed:
feat: v1alpha2 Paas validation webhook (#541)

* feat: migrate Paas validation webhook to v1alpha2

* test: add test suite for v1alpha2 Paas validation webhook

* feat: validate namespace groups in v1alpha2 Paas webhook

* feat(v1alpha2): add debug logging to conversion hook

devotional-phoenix-97 committed:
feat: removing paasns reconciler (#537)

* removing paasns controller

* Fixing some revive issues

* moving code to paasresource module

* adding namespaceDef

* Improving deletion without config

* Improved code

* Improving e2e tests. Dont wait for paasns if it is not created

* Improving e2e tests, use proper ssh secret

* making paasns e2e tests work

* Fixing unittests

* tuning coverage settings

* golangci-lint fixes

hikarukin committed:
feat: create PaasConfig v1alpha2 (#544)

* feat: create PaasConfig v1alpha2

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* remove deprecated stuff from PaasConfig type

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* initial basic docs adjustments

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* add storage and make manifests

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* re-add some deprecated entries for PaasConfig

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* temporarily adjust coverage thresholds

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* adjust PROJECT file

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* revert examples to v1alpha1

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

---------

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

devotional-phoenix-97 committed:
ci: bringing improving coverage for internal/controller/clusterrolebindings (#547)

* Improving coverage on internal/controller/clusterrolebindings

* Improving unittests on internal/controller/clusterrolebindings.go

devotional-phoenix-97 committed:
Improving of coverage on nsdefs (#557)

AxiomaticFixedChimpanzee committed:
Merge remote-tracking branch 'upstream/main' into v1alpha2

AxiomaticFixedChimpanzee committed:
test: add coverage for capabilities finalizer (#567)

devotional-phoenix-97 committed:
chore: rebase v1alpha1 on main (#581)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* Fixing shadow issues in v1alpha2

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

devotional-phoenix-97 committed:
moving all controller unittest funcs to suite_test.go (#573)

devotional-phoenix-97 committed:
chore: upstream (#583)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* unexporting getSchema (only required in internal/controller module) (#582)

* chore(deps): bump docker/build-push-action from 6.16.0 to 6.17.0 (#575)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/14487ce63c7a62a4a324b0bfb37086795e31c6c1...1dc73863535b631f98b2378be8619f83b136f4a0)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Defining assureAppSet (to be resued in other code in future PRs (#580)

Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* chore(deps): bump k8s.io/apimachinery from 0.33.0 to 0.33.1 (#576)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.0 to 0.33.1.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>

* chore(deps): bump k8s.io/apiextensions-apiserver from 0.33.0 to 0.33.1 (#577)

Bumps [k8s.io/apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver) from 0.33.0 to 0.33.1.
- [Release notes](https://github.com/kubernetes/apiextensions-apiserver/releases)
- [Commits](https://github.com/kubernetes/apiextensions-apiserver/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apiextensions-apiserver
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

devotional-phoenix-97 committed:
Improving of coverage on paas_controller (#560)

devotional-phoenix-97 committed:
chore: merge main to v1alpha2 (#590)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* unexporting getSchema (only required in internal/controller module) (#582)

* chore(deps): bump docker/build-push-action from 6.16.0 to 6.17.0 (#575)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/14487ce63c7a62a4a324b0bfb37086795e31c6c1...1dc73863535b631f98b2378be8619f83b136f4a0)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Defining assureAppSet (to be resued in other code in future PRs (#580)

Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* chore(deps): bump k8s.io/apimachinery from 0.33.0 to 0.33.1 (#576)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.0 to 0.33.1.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>

* chore(deps): bump k8s.io/apiextensions-apiserver from 0.33.0 to 0.33.1 (#577)

Bumps [k8s.io/apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver) from 0.33.0 to 0.33.1.
- [Release notes](https://github.com/kubernetes/apiextensions-apiserver/releases)
- [Commits](https://github.com/kubernetes/apiextensions-apiserver/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apiextensions-apiserver
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: Improve label selector to make methods less expensive (#572)

* Improve label selector to make methods less expensive

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Use secretList instead of []secret to reduce complexity.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix nil pointer

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Adds header back.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Lint fix

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Improving of coverage on paas_controller (#585)

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

portly-halicore-76 committed:
chore: Removes redundant finalizers (#591)

* Removes redundant finalizers by relying on cascading delete.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Degraded testCov

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

devotional-phoenix-97 committed:
ci: Adding api/v1alpha2/paasns_types_test (#535)

* Improving unittests on api/v1alpha2/paasns_types.go

* api/v1alpha2/paasns_types_test.go changes

* deleting stuff

---------

Co-authored-by: Hikarukin <156204784+hikarukin@users.noreply.github.com>

hikarukin committed:
feat: move to v1alpha2.PaasConfig using conversion for v1alpha1.PaasConfig (#588)

* wip

* wip

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix tests

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* wip

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix e2e tests

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix linter issues

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix linter issues

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* improve unit test for paasconfig

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix linter issues

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* migrate v1alpha1 paasconfig types test to v1alpha2 test set

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* update v1alpha2 suite test

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* Remove unneeded code

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* Remove unneeded namespace

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* v1alpha2 paasconfig webhook

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fix linter issues

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* register v2 webhook

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

---------

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

portly-halicore-76 committed:
fix: PaasConfig webhooks (#597)

* * Adds missing manifests for conversion webhook
* Adds patch to CRD
* Updates informer.go to use v1alpha2

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Adds missing manifests for conversion webhook
* Adds patch to CRD
* Updates informer.go to use v1alpha2
* Updates paasconfig_controller.go to v1alpha2 reconciler

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Adds missing manifests for conversion webhook
* Adds patch to CRD
* Updates informer.go to use v1alpha2
* Updates paasconfig_controller.go to v1alpha2 reconciler
* Updates paasconfig_controller.go Conditions to v1alpha2

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
fix: Removes predicate in Watched PaasNs, as that predicate is unrelated. (#596)

* Removes predicate in Watched PaasNs, as that predicate is unrelated.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Altered predicate to match Paas predicate

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

devotional-phoenix-97 committed:
chore: Adding update tests to paas reconcile unittests (#594)

* Adding paas update unittests

* Comments

---------

Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
feat!: No default namespace (#598)

* * Gets rid of the Default namespace named after the paas.
* Refactors namespace_def.go to be more reusable and clear
* Fixes management of secrets correctly in namespace_def.go
* Altered / added tests accordingly.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Gets rid of the Default namespace named after the paas.
* Refactors namespace_def.go to be more reusable and clear
* Fixes management of secrets correctly in namespace_def.go
* Altered / added tests accordingly.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes e2e-tests
* Fixes linting

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes linting for real

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes e2e for real

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes e2e for realz

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Changes paasNsReconciler order to make sure namespaces are finalized before reconciling secrets
* Removes functionally incorrect test from paas_controller_test.go. When a namespace is being terminated, all resources in it will be removed. As we update the paas in the test to remove all namespaces, we can assume all secrets are removed as well. In EnvTest this is rather difficult to test.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes thresholds for tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

AxiomaticFixedChimpanzee committed:
test(controller): add scenarios to cluster-wide quota logic (#593)

Stef16Robbe committed:
test(controller): add secrets controller tests (#584)

appiepollo14 committed:
Fixes https://github.com/belastingdienst/opr-paas/issues/586 (#605)

Signed-off-by: appiepollo14 <asjer94@live.com>

devotional-phoenix-97 committed:
chore: Adding PaasNs webhook with unittests (#602)

* Adding PaasNs webhook with unittests

* Fixes missing setup for PaasNs v1alpha2 webhook

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fixes lint issue

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Removes reference to ArgoCD cap from docs

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
fix: Unique rolebinding entries (#603)

* * Fixes https://github.com/belastingdienst/opr-paas/issues/595

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Resolves comments

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

devotional-phoenix-97 committed:
chore: merge main to v1alpha2 (#607)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* unexporting getSchema (only required in internal/controller module) (#582)

* chore(deps): bump docker/build-push-action from 6.16.0 to 6.17.0 (#575)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/14487ce63c7a62a4a324b0bfb37086795e31c6c1...1dc73863535b631f98b2378be8619f83b136f4a0)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Defining assureAppSet (to be resued in other code in future PRs (#580)

Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* chore(deps): bump k8s.io/apimachinery from 0.33.0 to 0.33.1 (#576)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.0 to 0.33.1.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>

* chore(deps): bump k8s.io/apiextensions-apiserver from 0.33.0 to 0.33.1 (#577)

Bumps [k8s.io/apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver) from 0.33.0 to 0.33.1.
- [Release notes](https://github.com/kubernetes/apiextensions-apiserver/releases)
- [Commits](https://github.com/kubernetes/apiextensions-apiserver/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apiextensions-apiserver
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: Improve label selector to make methods less expensive (#572)

* Improve label selector to make methods less expensive

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Use secretList instead of []secret to reduce complexity.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix nil pointer

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Adds header back.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Lint fix

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Improving of coverage on paas_controller (#585)

* chore(deps): bump github.com/gin-gonic/gin from 1.10.0 to 1.10.1 (#600)

Bumps [github.com/gin-gonic/gin](https://github.com/gin-gonic/gin) from 1.10.0 to 1.10.1.
- [Release notes](https://github.com/gin-gonic/gin/releases)
- [Changelog](https://github.com/gin-gonic/gin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/gin-gonic/gin/compare/v1.10.0...v1.10.1)

---
updated-dependencies:
- dependency-name: github.com/gin-gonic/gin
  dependency-version: 1.10.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump sigs.k8s.io/controller-runtime from 0.20.4 to 0.21.0 (#601)

Bumps [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from 0.20.4 to 0.21.0.
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.4...v0.21.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-version: 0.21.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* * Removes reference to ArgoCD cap from docs (#604)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/v1alpha2' into v1alpha2/release

devotional-phoenix-97 committed:
When deleting, controller updates the statusblock, which should not trigger update validations (#611)

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/main' into v1alpha2/release

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/v1alpha2' into v1alpha2/release

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/main' into v1alpha2/release

devotional-phoenix-97 committed:
Statusblock in PaasNS is obsolete (#613)

Stef16Robbe committed:
chore: v1alpha2 storageversion migration (#614)

* chore(api): move storageversion to v1alpha2

Signed-off-by: Stef Robbe <stef.robbe@protonmail.com>

* fix: Unique rolebinding entries (#603)

* * Fixes https://github.com/belastingdienst/opr-paas/issues/595

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Resolves comments

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* chore(conversion): swap hub/spoke conversion for paas

* chore(conversion): add hub/spoke conversion for paasns resources

* chore: merge main to v1alpha2 (#607)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* unexporting getSchema (only required in internal/controller module) (#582)

* chore(deps): bump docker/build-push-action from 6.16.0 to 6.17.0 (#575)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/14487ce63c7a62a4a324b0bfb37086795e31c6c1...1dc73863535b631f98b2378be8619f83b136f4a0)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Defining assureAppSet (to be resued in other code in future PRs (#580)

Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* chore(deps): bump k8s.io/apimachinery from 0.33.0 to 0.33.1 (#576)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.0 to 0.33.1.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>

* chore(deps): bump k8s.io/apiextensions-apiserver from 0.33.0 to 0.33.1 (#577)

Bumps [k8s.io/apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver) from 0.33.0 to 0.33.1.
- [Release notes](https://github.com/kubernetes/apiextensions-apiserver/releases)
- [Commits](https://github.com/kubernetes/apiextensions-apiserver/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apiextensions-apiserver
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: Improve label selector to make methods less expensive (#572)

* Improve label selector to make methods less expensive

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Use secretList instead of []secret to reduce complexity.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix nil pointer

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Adds header back.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Lint fix

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Improving of coverage on paas_controller (#585)

* chore(deps): bump github.com/gin-gonic/gin from 1.10.0 to 1.10.1 (#600)

Bumps [github.com/gin-gonic/gin](https://github.com/gin-gonic/gin) from 1.10.0 to 1.10.1.
- [Release notes](https://github.com/gin-gonic/gin/releases)
- [Changelog](https://github.com/gin-gonic/gin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/gin-gonic/gin/compare/v1.10.0...v1.10.1)

---
updated-dependencies:
- dependency-name: github.com/gin-gonic/gin
  dependency-version: 1.10.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump sigs.k8s.io/controller-runtime from 0.20.4 to 0.21.0 (#601)

Bumps [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from 0.20.4 to 0.21.0.
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.4...v0.21.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-version: 0.21.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* * Removes reference to ArgoCD cap from docs (#604)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* When deleting, controller updates the statusblock, which should not trigger update validations (#611)

* Statusblock in PaasNS is obsolete (#613)

* chore(api): swap spoke/hub conversion of PaasConfig

* fix(conversion): fixes for hub/spoke

---------

Signed-off-by: Stef Robbe <stef.robbe@protonmail.com>
Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

hikarukin committed:
feat: Controller conversion to v1alpha2 (#599)

* Initial conversion of controllers to v1alpha2

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* wip

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* wip

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* pleasure the linter

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* fixing some tests

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

* partially fix unit tests

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

---------

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

hikarukin committed:
wip

Signed-off-by: hikarukin <156204784+hikarukin@users.noreply.github.com>

portly-halicore-76 committed:
* Fixes broken tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
* Bump CI

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
Merge branch 'v1alpha2' into v1alpha2/release

portly-halicore-76 committed:
fix: failing v1alpha2 tests (#615)

* * Updates e2e test
* Fixes clusterwide quota
* Fixes non-clusterwide quota
* Implements quota deepcopy
* Fixes Linting issues
* Fixes broken tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* * Fixes decrypting?

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

Stef16Robbe committed:
test(conversion): add tests for paasns conversions (#616)

* test(conversion): add tests for paasns conversions

* fix(coverage): controller cov goal to 62

---------

Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

portly-halicore-76 committed:
Merge branch 'v1alpha2' into v1alpha2/release

devotional-phoenix-97 committed:
chore: merging v1alpha2 and main and merge into v1alpha2 if needed (#608)

* Based on the issue which was found in #568, we implement a linter to catch shadowing issues before they arise (#571)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* moving all controller unittest funcs to suite_test.go (#574)

* unexporting getSchema (only required in internal/controller module) (#582)

* chore(deps): bump docker/build-push-action from 6.16.0 to 6.17.0 (#575)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/14487ce63c7a62a4a324b0bfb37086795e31c6c1...1dc73863535b631f98b2378be8619f83b136f4a0)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.17.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Defining assureAppSet (to be resued in other code in future PRs (#580)

Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

* chore(deps): bump k8s.io/apimachinery from 0.33.0 to 0.33.1 (#576)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.0 to 0.33.1.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Devotional Phoenix <158271509+devotional-phoenix-97@users.noreply.github.com>

* chore(deps): bump k8s.io/apiextensions-apiserver from 0.33.0 to 0.33.1 (#577)

Bumps [k8s.io/apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver) from 0.33.0 to 0.33.1.
- [Release notes](https://github.com/kubernetes/apiextensions-apiserver/releases)
- [Commits](https://github.com/kubernetes/apiextensions-apiserver/compare/v0.33.0...v0.33.1)

---
updated-dependencies:
- dependency-name: k8s.io/apiextensions-apiserver
  dependency-version: 0.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: Improve label selector to make methods less expensive (#572)

* Improve label selector to make methods less expensive

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Use secretList instead of []secret to reduce complexity.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix nil pointer

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Adds header back.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Fix e2e tests

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Make a specific label for Paas to prevent duplicate labels passed on by ClonedLabel func.

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Lint fix

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* Improving of coverage on paas_controller (#585)

* chore(deps): bump github.com/gin-gonic/gin from 1.10.0 to 1.10.1 (#600)

Bumps [github.com/gin-gonic/gin](https://github.com/gin-gonic/gin) from 1.10.0 to 1.10.1.
- [Release notes](https://github.com/gin-gonic/gin/releases)
- [Changelog](https://github.com/gin-gonic/gin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/gin-gonic/gin/compare/v1.10.0...v1.10.1)

---
updated-dependencies:
- dependency-name: github.com/gin-gonic/gin
  dependency-version: 1.10.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump sigs.k8s.io/controller-runtime from 0.20.4 to 0.21.0 (#601)

Bumps [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from 0.20.4 to 0.21.0.
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.4...v0.21.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-version: 0.21.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* * Removes reference to ArgoCD cap from docs (#604)

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>

* When deleting, controller updates the statusblock, which should not trigger update validations (#612)

* chore(deps): bump docker/build-push-action from 6.17.0 to 6.18.0 (#610)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6.17.0 to 6.18.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/1dc73863535b631f98b2378be8619f83b136f4a0...263435318d21b8e681c14492fe198d362a7d2c83)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: 6.18.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

---------

Signed-off-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: portly-halicore-76 <170707699+portly-halicore-76@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: AxiomaticFixedChimpanzee <179703613+AxiomaticFixedChimpanzee@users.noreply.github.com>

portly-halicore-76 committed:
Merge remote-tracking branch 'upstream/v1alpha2' into v1alpha2/release

hikarukin reviewed (changes requested):
Needs more testing.

Paas.Spec.Namespaces[ns].Secrets is not being honored from the look of it.
@nexus-core174 is writing some additional tests in that regard.

Do not merge - Solved

devotional-phoenix-97 committed:
chore: Adding code and unittests for managing groups and secrets in paas.Spec.Namespaces{} (#617)

* Adding unittests for groups and secrets in paas.Spec.Namespaces{}

* Adding code for managing groups and secrets in paas.Spec.Namespaces{}

* Fixing comment from Portly Halicore

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/v1alpha2' into v1alpha2/release

devotional-phoenix-97 committed:
fix: ArgoCD obsolete fields during conversion (#618)

* when converting v1alpha2 to v1alpha1, use fields, dont use obsolete fields for argocd

* Having both custom fields and obsolete fields in v1alpha1 is probably best (e.a. to prevent ArgoCD missing fields that it is managing).

* Fix broken test

assertV1Conversion is dependent on the update done in assertV2Created

Signed-off-by: Hikarukin <156204784+hikarukin@users.noreply.github.com>

* pleasure the linter

Signed-off-by: Hikarukin <156204784+hikarukin@users.noreply.github.com>

---------

Signed-off-by: Hikarukin <156204784+hikarukin@users.noreply.github.com>
Co-authored-by: Hikarukin <156204784+hikarukin@users.noreply.github.com>

devotional-phoenix-97 committed:
Merge remote-tracking branch 'bd/v1alpha2' into v1alpha2/release
